DOCUMENTATION.md

```diff
@@ -104,751 +29,740 @@ import uk.gov.service.notify.SendSmsResponse;
 
-### Letter
+### Method
 
-#### Method
+```java
+SendLetterResponse response = client.sendLetter(templateId, personalisation, reference);
+```
+
+### Arguments
 
-The letter must contain:
+#### templateId (required)
 
-- mandatory address fields
-- optional address fields if applicable
-- fields from template
+The ID of the template. You can find this by signing into GOV.UK Notify and going to the __Templates__ page.
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+```
+String templateId="f33517ff-2a88-4f6e-b855-c550268ce08a";
+```
+
+#### personalisation (required)
+
+The personalisation argument always contains the following parameters for the letter recipient's address:
+
+- `address_line_1`
+- `address_line_2`
+- `postcode`
+
+Any other placeholder fields included in the letter template also count as required parameters. You need to provide their values in a map. For example:
 
 ```java
 HashMap<String, String> personalisation = new HashMap<>();
 personalisation.put("address_line_1", "The Occupier"); // mandatory address field
 personalisation.put("address_line_2", "Flat 2"); // mandatory address field
+personalisation.put("postcode", "SW14 6BH"); // mandatory address field
+personalisation.put("first_name", "Amala"); // field from template
+personalisation.put("application_date", "2018-01-01"); // field from template
+```
+
+If a template does not have any placeholder fields for personalised information, you must pass in an empty map or `null`.
+
+#### personalisation (optional)
+
+The following parameters in the letter recipient's address are optional:
+
+```java
 personalisation.put("address_line_3", "123 High Street"); // optional address field
 personalisation.put("address_line_4", "Richmond upon Thames"); // optional address field
 personalisation.put("address_line_5", "London"); // optional address field
 personalisation.put("address_line_6", "Middlesex"); // optional address field
-personalisation.put("postcode", "SW14 6BH"); // mandatory address field
-personalisation.put("application_id", "1234"); // field from template
-personalisation.put("application_date", "2017-01-01"); // field from template
-
-SendLetterResponse response = client.sendLetter(templateId, personalisation, "yourReferenceString");
 ```
-</details>
 
-#### Response
+#### reference (optional)
+
+A unique identifier. This reference can identify a single unique notification or a batch of multiple notifications. If you do not have a reference, you must pass in an empty string or `null`.
 
-If the request is successful, the SendLetterResponse is returned from the client. Attributes of the SendLetterResponse are listed below.
+```
+String reference='STRING';
+```
+
+### Response
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+If the request to the client is successful, you will receive a `SendLetterResponse`:
```

Comment:
Maybe this will read better as:
... you will receive a `SendLetterResponse` with the following attributes:

Reply:
Changed